apps/prairielearn/src/lib/editors.js

```diff
@@ -1386,30 +1386,21 @@ export class FileModifyEditor extends Editor {
 export class CourseInfoEditor extends Editor {
   constructor(params) {
     super(params);
-    this.description = `Create infoCourse.json`;
+    this.description = `Write infoCourse.json`;
+    this.infoJson = params.infoJson;
+    this.flag = params.flag || '';
```

Comment:
Instead of using a new subclass, I reused the `CourseInfoEditor` subclass. We then differentiate if we are adding a new `infoCourse.json` or reading the current file and overwriting the changes in the controller. Those settings are then passed through the params here.

Reply:
Sorry about that! Now using the `FileModifyEditor` in [79f4f05](https://github.com/PrairieLearn/PrairieLearn/pull/9581/commits/79f4f05b8d78db39a94b29e232a22bdc4c552b34).